nautobot/docs/guides/graphql.md

```diff
@@ -0,0 +1,1041 @@
+
+```
+query {
+  devices {
+    name
+    interfaces {
+      name
+      description
+      mgmt_only
+      ip_addresses {
+        address
+      }   
+    }
+  }
+}
+```
+
+The results of the query look like:
+
+![Graph*i*QL - Simple Query 2](/guides/images/graphql/graphiql-07.png)
+
+### Filtering Queries
+
+These queries are great, but they are displaying the interface attributes and device names for every device in the Nautobot inventory. Currently, Nautobot allows users to filter queries at the top level of the query. In our previous examples, the top level would be the `devices` query.
+
+As an example. We can query devices by their site location. This is done by added `(site: "<site name>")` after devices. For example: `query { devices(site: "ams") { name }}` will display all devices in the `ams` site.
+
+```
+{
+  "data": {
+    "devices": [
+      {
+        "name": "ams-edge-01"
+      },
+      {
+        "name": "ams-edge-02"
+      },
+      {
+        "name": "ams-leaf-01"
+      },
+      {
+        "name": "ams-leaf-02"
+      },
+      {
+        "name": "ams-leaf-03"
+      },
+      {
+        "name": "ams-leaf-04"
+      },
+      {
+        "name": "ams-leaf-05"
+      },
+      {
+        "name": "ams-leaf-06"
+      },
+      {
+        "name": "ams-leaf-07"
+      },
+      {
+        "name": "ams-leaf-08"
+      }
+    ]
+  }
+}
+```
+
+Graph*i*QL allows you to add multiple attributes to the filter criteria. You can use the `Documentation Explorer` to assist you in finding criteria attributes to filte on. In this example, I add the `manufacturer` attribute in addition to `site`. 
+
+```
+query {
+  devices(site:"ams", manufacturer: "Arista") {
+    name
+  }
+}
+```
+
+```
+{
+  "data": {
+    "devices": [
+      {
+        "name": "ams-edge-01"
+      },
+      {
+        "name": "ams-edge-02"
+      },
+      {
+        "name": "ams-leaf-01"
+      },
+      {
+        "name": "ams-leaf-02"
+      },
+      {
+        "name": "ams-leaf-03"
+      },
+      {
+        "name": "ams-leaf-04"
+      },
+      {
+        "name": "ams-leaf-05"
+      },
+      {
+        "name": "ams-leaf-06"
+      },
+      {
+        "name": "ams-leaf-07"
+      },
+      {
+        "name": "ams-leaf-08"
+      }
+    ]
+  }
+}
+```
+
+## Using the GraphQL API in Nautobot
+
+Now that we've explored how to use the Graph*i*QL interface to help us create GraphQL queries, let's take our queries and call them with the Rest API. This is where the real advantage is going to come in to play, because it will allow us to utilize these queries in a programmatic way.
```

Comment:
`Rest` -> `REST` throughout please.

```suggestion
Now that we've explored how to use the Graph*i*QL interface to help us create GraphQL queries, let's take our queries and call them with the REST API. This is where the real advantage is going to come in to play, because it will allow us to utilize these queries in a programmatic way.
```